invenio_rdm_records/services/permissions.py

```diff
@@ -82,9 +83,17 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
         IfRestricted("record", then_=can_view, else_=can_all),
     ]
     # Allow reading the files of a record
-    can_read_files = [
+    # Hide record's files from community
+    can_read_files_owner = [RecordOwners(), SecretLinks("view"), SystemProcess()]
+    # Show record's files to community
+    can_read_files_community = [
         IfRestricted("files", then_=can_view, else_=can_all),
     ]
+    can_read_files = [
+        IfRestrictedForCommunity(
```

Comment:
open question for reviewers (we discussed with it with Anika):

Should all this go to ZenodoRDM only? it is possible thanks to the config var [RDM_PERMISSION_POLICY](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/config.py#L105). However, you will have to main the list of permissions updated in ZenodoRDM if changes.

@slint do you prefer to have this only in ZenodoRDM?

Reply:
Yes, this should go in ZenodoRDM only, so that we keep the base permission policy class minimal. We can put it in `invenio.cfg` like so:

```python
# in zenodo-rdm/invenio.cfg

class ZenodoRDMRecordPermissionPolicy(RDMRecordPermissionPolicy):

    # Allow reading the files of a record
    # Hide record's files from community
    can_read_files_owner = [RecordOwners(), SecretLinks("view"), SystemProcess()]
    # Show record's files to community
    can_read_files_community = [
        IfRestricted(
            "files", 
            then_=RDMRecordPermissionPolicy.can_view, 
            else_=RDMRecordPermissionPolicy.can_all
        ),
    ]
    can_read_files = [
        IfFilesRestrictedForCommunity(
            then_=can_read_files_owner, else_=can_read_files_community
        ),
    ]
    can_get_content_files = [
        # note: even though this is closer to business logic than permissions,
        # it was simpler and less coupling to implement this as permission check
        IfFileIsLocal(then_=can_read_files, else_=[SystemProcess()])
    ]

RDM_PERMISSION_POLICY = ZenodoRDMRecordPermissionPolicy
```

The inheritance/reusability is a bit broken/tricky in here, e.g. some attributes like `can_get_content_files` have to be redefined also since their dependent attributes are redefined in the class... Mangling the parent class fields is probably the best we can do now in terms of maintainability, so let's go with this and see in the future if we need something smarter as discussed in the past.